### ISSUE TITLE
multilabel accuracy is lower in 2.2.0 than 2.1.0 for same code

### ISSUE THREAD

**Issue by amaiya** [tensorflow/tensorflow#41114]
The following simple multilabel classification example returns a **100% accuracy in TensorFlow 2.1.0** but only **60%  accuracy in TensorFlow 2.2.0**.  The loss appears to be the same, though.    **Why is this?** 

```python
X = [[1,0,0,0,0,0,0],
      [1,2,0,0,0,0,0],
      [3,0,0,0,0,0,0],
      [3,4,0,0,0,0,0],
      [2,0,0,0,0,0,0],
      [3,0,0,0,0,0,0],
      [4,0,0,0,0,0,0],
      [2,3,0,0,0,0,0],
      [1,2,3,0,0,0,0],
      [1,2,3,4,0,0,0],
      [0,0,0,0,0,0,0],
      [1,1,2,3,0,0,0],
      [2,3,3,4,0,0,0],
      [4,4,1,1,2,0,0],
      [1,2,3,3,3,3,3],
      [2,4,2,4,2,0,0],
      [1,3,3,3,0,0,0],
      [4,4,0,0,0,0,0],
      [3,3,0,0,0,0,0],
      [1,1,4,0,0,0,0]]

Y = [[1,0,0,0],
    [1,1,0,0],
    [0,0,1,0],
    [0,0,1,1],
    [0,1,0,0],
    [0,0,1,0],
    [0,0,0,1],
    [0,1,1,0],
    [1,1,1,0],
    [1,1,1,1],
    [0,0,0,0],
    [1,1,1,0],
    [0,1,1,1],
    [1,1,0,1],
    [1,1,1,0],
    [0,1,0,0],
    [1,0,1,0],
    [0,0,0,1],
    [0,0,1,0],
    [1,0,0,1]]
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense
from tensorflow.keras.layers import Embedding
from tensorflow.keras.layers import GlobalAveragePooling1D
import numpy as np


X = np.array(X)
Y = np.array(Y)
MAXLEN = 7
MAXFEATURES = 4
NUM_CLASSES = 4
model = Sequential()
model.add(Embedding(MAXFEATURES+1,
                    50,
                    input_length=MAXLEN))
model.add(GlobalAveragePooling1D())
model.add(Dense(NUM_CLASSES, activation='sigmoid'))
model.compile(loss='binary_crossentropy',
              optimizer='adam',
              metrics=['accuracy'])
model.fit(X, Y,
          batch_size=1,
          epochs=200,
          validation_data=(X, Y))
```

**Comment by Saduf2019** [comment]
I ran the code shared, please find the accuracy for [tf 2.2](https://colab.research.google.com/gist/Saduf2019/c8ce0b0ca9528248a67571c3f0d574d3/untitled259.ipynb) [tf 2.1](https://colab.research.google.com/gist/Saduf2019/720f9afebd68b6d6f89c6564282f23b6/untitled256.ipynb)

**Comment by jvishnuvardhan** [comment]
@amaiya I think this is related to a know issue. We can track the progress [here](https://github.com/tensorflow/tensorflow/issues/42045).

Simple workaround is to change `'accuracy'` to `'binary_accuracy'`. When I modified your code, everything working well in TF2.3 similar to TF2.1. [Here](https://colab.research.google.com/gist/jvishnuvardhan/0a8f1a05f8d5dd55a8539cad07a99500/untitled256.ipynb) is the gist for your reference. Thanks!

**Comment by amaiya** [comment]
Thanks, @jvishnuvardhan  

I confirm that everything works when changing `accuracy` to `binary_accuracy` in both TF 2.1 and TF 2.2.  As this was [indicated to be a bug](https://github.com/tensorflow/tensorflow/issues/42045#issuecomment-674232499), I assume the old behavior of `accuracy` will return in TF 2.4.

**Comment by jvishnuvardhan** [comment]
@amaiya I will close this issue and track the progress in that issue I mentioned in last response. Thanks!

**Comment by jvishnuvardhan** [comment]
This is resolved in recent tf-nightly. Please feel free to reopen if you notice the issue.
This is available in stable TF2.4 in the near future. Thanks!

**Comment by amaiya** [comment]
@jvishnuvardhan I just tested this with `tf-nightly` and it is still broken when using `accuracy`. Thanks.
